sqlite3/lib/src/ffi/constants.dart

```diff
@@ -1,44 +1,247 @@
+  static const int SQLITE_ABORT = 4;
+
+  /// The database file is locked
+  static const int SQLITE_BUSY = 5;
+
+  /// A table in the database is locked
+  static const int SQLITE_LOCKED = 6;
+
+  /// A malloc() failed
+  static const int SQLITE_NOMEM = 7;
+
+  /// Attempt to write a readonly database
+  static const int SQLITE_READONLY = 8;
+
+  /// Operation terminated by sqlite3_interrupt()
+  static const int SQLITE_INTERRUPT = 9;
+
+  /// Some kind of disk I/O error occurred
+  static const int SQLITE_IOERR = 10;
+
+  /// The database disk image is malformed
+  static const int SQLITE_CORRUPT = 11;
+
+  /// Unknown opcode in sqlite3_file_control()
+  static const int SQLITE_NOTFOUND = 12;
+
+  /// Insertion failed because database is full
+  static const int SQLITE_FULL = 13;
+
+  /// Unable to open the database file
+  static const int SQLITE_CANTOPEN = 14;
+
+  /// Database lock protocol error
+  static const int SQLITE_PROTOCOL = 15;
+
+  /// Internal use only
+  static const int SQLITE_EMPTY = 16;
+
+  /// The database schema changed
+  static const int SQLITE_SCHEMA = 17;
+
+  /// String or BLOB exceeds size limit
+  static const int SQLITE_TOOBIG = 18;
+
+  /// Abort due to constraint violation
+  static const int SQLITE_CONSTRAINT = 19;
+
+  /// Data type mismatch
+  static const int SQLITE_MISMATCH = 20;
+
+  /// Library used incorrectly
+  static const int SQLITE_MISUSE = 21;
+
+  /// Uses OS features not supported on host
+  static const int SQLITE_NOLFS = 22;
+
+  /// Authorization denied
+  static const int SQLITE_AUTH = 23;
+
+  /// Not used
+  static const int SQLITE_FORMAT = 24;
+
+  /// 2nd parameter to sqlite3_bind out of range
+  static const int SQLITE_RANGE = 25;
+
+  /// File opened that is not a database file
+  static const int SQLITE_NOTADB = 26;
+
+  /// Notifications from sqlite3_log()
+  static const int SQLITE_NOTICE = 27;
+
+  /// Warnings from sqlite3_log()
+  static const int SQLITE_WARNING = 28;
 
-// Flags for file open operations, https://www.sqlite.org/c3ref/c_open_autoproxy.html
+  /// sqlite3_step() has another row ready
+  static const int SQLITE_ROW = 100;
 
-const SQLITE_OPEN_READONLY = 0x00000001;
-const SQLITE_OPEN_READWRITE = 0x00000002;
-const SQLITE_OPEN_CREATE = 0x00000004;
-const SQLITE_OPEN_URI = 0x00000040;
-const SQLITE_OPEN_MEMORY = 0x00000080;
-const SQLITE_OPEN_NOMUTEX = 0x00008000;
-const SQLITE_OPEN_FULLMUTEX = 0x00010000;
-const SQLITE_OPEN_SHAREDCACHE = 0x00020000;
-const SQLITE_OPEN_PRIVATECACHE = 0x00040000;
-const SQLITE_OPEN_NOFOLLOW = 0x01000000;
+  /// sqlite3_step() has finished executing
+  static const int SQLITE_DONE = 101;
+}
+
+/// Flags for file open operations, https://www.sqlite.org/c3ref/c_open_autoproxy.html
+/// Flags For File Open Operations
+///
+/// These bit values are intended for use in the
+/// 3rd parameter to the [sqlite3_open_v2()] interface and
+/// in the 4th parameter to the [sqlite3_vfs.xOpen] method.
+class SqlFlags {
```

Comment:
Do we need to add those flags to the public api? I think we don't provide an api where users could pass one of them, so I think we shouldn't make them public.

Reply:
You are right, I've changed the visibility in sqlite3.dart on the last commit